p2p/src/authenticated/actors/peer/actor.rs

```diff
@@ -61,25 +63,35 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng> Actor<E> {
         )
     }
 
-    async fn send_content<Si: Sink>(
+    // prepare_msg returns Data after checking its channel is contained by rate_limits.
```

Comment:
nits: These function comments should be `///` rather than `//`. In Rust, it also isn't required to start the comment with the function name.

After a little more thought, I think the function should actually be `validate_msg(msg: Option<wire::Data>,...)` and the comment should be `Unpack msg and verify the channel it is over is registered` (or something like that).

Reply:
If you wanted to go for "bonus points", you could another comment indicating:
```
///
/// The provided `rate_limits` is not updated during this function and is just used to verify the data's channel is registered.
```

I suppose that is implicit from the non-mutable reference but 🤷 .

Reply:
I have done a mix, let me know if it's ok 🙂